### PR TITLE
custom exception handler

### PR DESCRIPTION
**Pull request by jgillich**

### CODE REVIEW

File: src/graphql/context.cr [hunk 1]
```diff
@@ -2,14 +2,16 @@ require "./language"
...
     property fragments : Array(Language::FragmentDefinition) = [] of Language::FragmentDefinition
     property query_type : String = ""
     property mutation_type : String? = nil
     property document : Language::Document?
 
-    def initialize(@max_complexity = nil)
+    # Return string message to be added to errors object or throw to bubble up
+    def handle_exception(ex : Exception) : String?
+      ex.message
     end
```
Comment (jwoertink) [inline src/graphql/context.cr:15]: oh, clever... So we would just override this in our own context, then we could propagate whichever error we want? I like that.

File: src/graphql/object_type.cr [hunk 2]
```diff
@@ -46,9 +46,10 @@ module GraphQL::ObjectType
             begin
               errors.concat _graphql_resolve(context, selection, json)
             rescue e
-              if !e.message.nil?
+              message = context.handle_exception(e)
+              if !message.nil?
```
Comment (jwoertink) [inline src/graphql/object_type.cr:50]:
```suggestion
              if message = context.handle_exception(e)
```

This is a pretty standard Crystal convention, and you wouldn't need to use the `not_nil!` below

File: spec/exception_spec.cr [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          err
+        }
+      ),
+      context: ExceptionFixture::Context.new
+    ).should eq (
+      {
+        "data"   => {} of Nil => Nil,
+        "errors" => [
+          {
+            "message" => "handled",
```
Comment (jwoertink) [inline spec/exception_spec.cr:33]: :heart:

File: src/graphql/object_type.cr (hunk 2 above)
Comment (jgillich) [reply src/graphql/object_type.cr]: Nice, thanks

**Pull request merged**